openapi_python_client/parser/properties.py

```diff
@@ -64,20 +64,22 @@ def get_imports(self, *, prefix: str) -> Set[str]:
             back to the root of the generated client.
         """
         if self.nullable or not self.required:
-            return {"from typing import Optional"}
+            return {"from typing import cast, Optional", f"from {prefix}types import UNSET, Unset"}
```

Comment:
I think you won't need cast anymore if you take my suggestion of setting the type with Union. You will, however, have to import Union.

```suggestion
            return {"from typing import Union, Optional", f"from {prefix}types import UNSET, Unset"}
```